lib/spack/spack/cmd/__init__.py

```diff
@@ -267,39 +269,48 @@ def matching_specs_from_env(specs):
     return _concretize_spec_pairs(spec_pairs + additional_concrete_specs)[: len(spec_pairs)]
 
 
-def disambiguate_spec(spec, env, local=False, installed=True, first=False):
+def disambiguate_spec(
+    spec: "spack.spec.Spec",
```

Comment:
can you drop all the `"`s? That shouldn't be necessary in cmd/*

Reply:
Should be done